src/Serializer/AbstractItemNormalizer.php

```diff
@@ -298,6 +301,17 @@ private function denormalizeRelation(string $attributeName, PropertyMetadata $pr
         }
 
         if (!is_array($value)) {
+            // repeat the code so that IRIs keep working with the json format
+            if ('json' === $format && $this->itemDataProvider) {
```

Comment:
I get the idea and the use case, but it hurts me a bit:

* it doesn't fit the hypermedia philosophie of API Platform
* it doesn't work consistently across formats (XML or CSV for instance may need this feature as well)

May I suggest an alternative implementation:

* add a new flag `allow_plain_identifiers` set to false by default
* if this flag is on, execute your code in all circonstances, regardless of the format

Reply:
And then it will be merged in master.

Reply:
You see :p had the same concerns ^^